core/blockchain.go

```diff
@@ -2820,8 +2703,12 @@ func (bc *BlockChain) UpdateStakingMetaData(tx *staking.StakingTransaction, root
 		if err != nil {
 			return err
 		}
-		err = rawdb.WriteValidatorSnapshot(bc.db, validator)
-		if err != nil {
+		zero := big.NewInt(0)
+		validator.Snapshot.Epoch = epoch
```

Comment:
don't need to modify this. When the wrapper is created. you already populated these fields with 0. Should just copy and record the wrapper as is.

Reply:
gotcha, so removed the two counters but i think i should still keep `validator.Snapshot.Epoch = epoch`

Reply:
why add an epoch? Then it's not really a snapshot anymore.